### ISSUE TITLE
Abandons Need to Add Host Info

### ISSUE THREAD

**Issue by abombss** [Particular/NServiceBus.AzureServiceBus#143]
https://github.com/Particular/NServiceBus.AzureServiceBus/blob/develop/src/Transport/Receiving/MessageReceiverNotifier.cs#L194

All calls to abandon should first try and add some host info to the brokered message properties.  This ensures if the MaxDeliveryAttempts is ever exceeded and the message is DLQ'd we can trace back to what endpoint the message came from.  This could also allow us to track e2e processing time which we currently can't do when messages are being abandoned and retried.

Some properties I would like to see added.
ProcessingEndpoint = so we know where it logically came from
ProcessingHostId = so we know physically which host this happened on
Reason = so we know why it was abandoned
FirstReceiveTime = to track e2e processing time first receive
Exception Details if there were any

Connects to https://github.com/Particular/NServiceBus.Azure/issues/310

**Comment by yvesgoeleven** [comment]
Good suggestion! Depends on https://github.com/Particular/NServiceBus/issues/3406 before we can add it though.

**Comment by tmasternak** [comment]
@yvesgoeleven This does not seem to be a must have for the V6 release. Since we can add additional information in a minor release. Removing the `V6 Launch` label.

**Comment by SeanFeldman** [comment]
@tmasternak we have [another issue](https://github.com/Particular/NServiceBus.AzureServiceBus/issues/142) that needs the same info, that one is left for `v6 Launch` when this issue was removed. Why are they treated differently?

**Comment by yvesgoeleven** [comment]
@tmasternak I take it that there is another process change that I'm unaware off? I would really appreciate it if we are at least consulted before our issues are being modified by other people. Changing states & tags without asking will only slow us down.

**Comment by andreasohlund** [comment]
@seanfeldman lets remove the label from that one as well or would you say its a must have for release?

**Comment by SeanFeldman** [comment]
@andreasohlund removing tags w/o consulting is wrong. We might not add all the information in `v6-launch`, but we could split the work (add minimal info that allows to troubleshoot where the message was abandoned in `v6-launch` and open a separate issue for improvement later, post `v6-launch`).

Quoting what the customer said: "If the DLQ is setup to forward to a global error queue, you have no idea what endpoint this message failed on."

**Comment by andreasohlund** [comment]
fair enough. Note that I'm not saying to not do it in v6, if we have time to add more features in we can discuss if its a priority to add but right now the focus is on the must haves so that we can get into a shippable state. (we're doing the same grooming on all repos not just this one)

**Comment by SeanFeldman** [comment]
@abombss do you think this is still needed? With NServiceBus 6 (ASB 7) we've [changed how retries are working](https://docs.particular.net/transports/azure-service-bus/retries#immediate-retries-and-maxdeliverycount) and connected it to the `MaxDeliveryCount` ensuring messages are only DLQ-ed when are physically corrupted.

> In versions 6 and below, the default value for MaxDeliveryCount is 6. Starting from Versions 7 and above, the new defaults for MaxDeliveryCount is one more than the configured value of Immediate Retries. The default value MaxDeliveryCount for system queues such as audit and error queues is 10. If the endpoint disables the Immediate Retries explicitly, then the MaxDeliveryCount will also be set to 10.

**Comment by SeanFeldman** [comment]
Given that this is a legacy transport and no new feature will be added, going to close it.

Also, hosting information should be provided via some sort of API applicable to all transports compatible with NServiceBus. Would be a good feature request for the [Core](https://github.com/Particular/NServiceBus) project.
